src/main/java/org/sagebionetworks/bridge/hibernate/HibernatePermissionDao.java

```diff
@@ -0,0 +1,101 @@
+package org.sagebionetworks.bridge.hibernate;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import org.sagebionetworks.bridge.dao.PermissionDao;
+import org.sagebionetworks.bridge.models.permissions.Permission;
+import org.sagebionetworks.bridge.models.permissions.EntityType;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class HibernatePermissionDao implements PermissionDao {
+    
+    static final String APP_ID = "appId";
+    static final String USER_ID = "userId";
+    static final String ENTITY_TYPE = "entityType";
+    static final String ENTITY_ID = "entityId";
+    
+    static final String GET_BY_USER = "FROM Permission WHERE appId=:appId AND userId=:userId";
+    static final String GET_BY_ENTITY_TYPE = "FROM Permission WHERE appId=:appId "+
+            "AND entityType=:entityType AND entityId=:entityId";
+    
+    private HibernateHelper hibernateHelper;
+    
+    @Resource(name = "mysqlHibernateHelper")
+    final void setHibernateHelper(HibernateHelper hibernateHelper) {
+        this.hibernateHelper = hibernateHelper;
+    }
+    
+    @Override
+    public Optional<Permission> getPermission(String appId, String guid) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(guid));
+        
+        Permission permission = hibernateHelper.getById(Permission.class, guid);
+        if (permission == null) {
+            return Optional.empty();
+        }
+        
+        return Optional.of(permission);
+    }
+    
+    @Override
+    public List<Permission> getPermissionsForUser(String appId, String userId) {
+        checkArgument(isNotBlank(appId));
+        checkArgument(isNotBlank(userId));
+        
+        QueryBuilder builder = new QueryBuilder();
+        builder.append(GET_BY_USER, APP_ID, appId, USER_ID, userId);
+        
+        return hibernateHelper.queryGet(
+                builder.getQuery(), builder.getParameters(), null, null, Permission.class);
```

Comment:
This is simple enough you might consider dropping QueryBuilder and just using the string constant and an ImmutableMap.of(...)